File: src/main/java/com/cfgglobal/test/domain/Permission.java
package com.cfgglobal.test.domain;


import lombok.*;
import lombok.experimental.Accessors;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.Entity;

@Entity
@Getter
@Setter
@ToString
@Accessors
@DynamicUpdate
@DynamicInsert
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Permission extends BaseEntity {

    @NonNull
    String entity;

    @NonNull
    String authKey;

    @NonNull
    String httpMethod;

    @NonNull
    String authUris;


    String icon;

    String menuUrl;

}

File: src/main/kotlin/com/cfgglobal/test/domain/Permission.kt
package com.cfgglobal.test.domain


import lombok.NonNull
import org.hibernate.annotations.DynamicInsert
import org.hibernate.annotations.DynamicUpdate
import javax.persistence.Entity

@Entity
@DynamicUpdate
@DynamicInsert
data class Permission(

        @NonNull
        var entity: String = "",

        @NonNull
        var authKey: String = "",

        @NonNull
        var httpMethod: String = "",

        @NonNull
        var authUris: String = ""

) : BaseEntity()
